include/parakeet/ProE/internal/Parser.h

```diff
@@ -0,0 +1,139 @@
+/*
+	Copyright 2021 OpenJAUS, LLC (dba MechaSpin). Subject to the MIT license.
+*/
+
+#ifndef PARAKEET_PROE_PARSER_H
+#define PARAKEET_PROE_PARSER_H
+
+#include <functional>
+#include <memory>
+#include <vector>
+#include <chrono>
+
+namespace mechaspin
+{
+namespace parakeet
+{
+namespace ProE
+{
+namespace internal
+{
+class Parser
+{
+	public:
+		struct LidarSensorProperties
+		{
+			bool unitIsInCM;
+			bool withIntensity;
+			bool doDragPointRemoval;
+			bool doDataSmoothing;
+
+			uint32_t value;
+		};
+
+		struct LidarPoint
+		{
+			uint16_t distance;
+			uint16_t relativeStartAngle;
+			uint8_t intensity;
+		};
+
+		struct CompleteLidarMessage
+		{
+			uint16_t totalPoints;
+			uint32_t startAngle;
+			uint32_t endAngle;
+
+			LidarSensorProperties sensorPropertyFlags;
+
+			std::chrono::system_clock::time_point timestamp;
+			uint32_t deviceNumber;
+
+			std::vector<LidarPoint> lidarPoints;
+		};
+
+		Parser(std::function<void(CompleteLidarMessage*)> onCompleteLidarMessageCallback);
+
+		int parseSensorBuffer(int length, unsigned char* buf);
```

Comment:
I'd probably just call this method parse. For more context I'd probably change the class name to MessageParser or CommandParser or something. For the UdpSocket class we have the char* parameter first and the length second, where here we have the length first and the char* second. I think it'd be good for it to be consistent. Possibly speaks to it being useful to hav a simple Data or Buffer type struct that groups the buffer and length together. Seems like that could be used in multiple places.